core/commands/dag/dag.go

```diff
@@ -241,3 +248,103 @@ var DagResolveCmd = &cmds.Command{
 	},
 	Type: ResolveOutput{},
 }
+
+var DagExportCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Streams the selected DAG as a .car stream on stdout.",
+		ShortDescription: `
+'ipfs dag export' fetches a dag and streams it out as a well-formed .car file.
+Note that at present only single root selections / .car files are supported.
+The output of blocks happens in strict DAG-traversal, first-seen, order.
+`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("root", true, false, "Expression evaluting to a single root of a dag to export").EnableStdin(),
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+
+		c, err := cid.Decode(req.Arguments[0])
+		if err != nil {
+			return fmt.Errorf(
+				"unable to parse root specification (currently only bare CIDs are supported): %s",
+				err,
+			)
+		}
+
+		// The current interface of go-car is rather suboptimal as it
+		// only takes a blockstore, instead of accepting a dagservice,
+		// and leveraging parallel-fetch capabilities
+		// https://github.com/ipld/go-car/issues/27
+		//
+		// Until the above is fixed, pre-warm the blockstore before doing
+		// anything else. We explicitly *DO NOT* take a lock during this
+		// operation: even if we lose some of the blocks we just received
+		// due to a conflicting GC: we will just re-retrieve anything we
+		// potentially lost when the car is being streamed out
+		node, err := cmdenv.GetNode(env)
+		if err != nil {
+			return err
+		}
+
+		if err := mdag.FetchGraph(req.Context, c, node.DAG); err != nil {
```

Comment:
@Stebalien in one of our earlier conversations you asked me to do this pre-fetch before using the NewSession thing below ( it's commented out at present, but ignore that )

During the pre-fetch the CLI remains silent for minutes, pretty crappy UX. Could we simply not run this prefetch, and let the car writer pull stuff out, provided https://github.com/ipld/go-car/issues/27 gets implemented instead...?

Reply:
I agree, and this will actually slow us down in the common case where we already have the data. The real solution is to teach the selector engine to prefetch (or at least hint that something needs to be loaded), but we can always do that later.

Reply:
gone